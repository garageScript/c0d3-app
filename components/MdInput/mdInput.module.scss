.textarea {
  padding: 0.5rem;
  width: 100%;
  resize: vertical;
  min-height: 100px;
  max-height: 500px;
}

.preview {
  padding: 0.5rem;
  width: 100%;
  display: block;
  width: inherit;
  height: auto;
}
