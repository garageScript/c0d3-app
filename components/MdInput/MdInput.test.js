import { mockUseIsMac } from '../../__mocks__/useIsMac.mock'
import '../../__mocks__/useBreakpoint.mock'
import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'

jest.mock('../../helpers/useBreakpoint.tsx')

import useBreakpoint from '../../helpers/useBreakpoint.tsx'
import MdInput from './MdInput'

const TestComponent = () => {
  const [testState, setTestState] = React.useState('')
  return <MdInput onChange={setTestState} value={testState} />
}

describe('MdInput Component', () => {
  test('Should render text onto textbox when user types', async () => {
    const { container } = render(<TestComponent />)

    await userEvent.type(screen.getByRole('textbox'), 'Javascript')
    expect(screen.getByRole('textbox')).toHaveValue('Javascript')
    expect(container).toMatchSnapshot()
  })

  test('Should switch to Write mode when user clicks Write button', async () => {
    const { container } = render(<MdInput />)

    await userEvent.click(screen.getByRole('button', { name: 'Write' }))

    expect(screen.queryByTestId('markdown')).not.toBeInTheDocument()
    expect(screen.getByRole('textbox')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  test("Should display 'Nothing to preview' when switching to Preview mode with no input", async () => {
    render(<MdInput />)

    await userEvent.click(screen.getByRole('button', { name: 'Preview' }))

    expect(screen.getByRole('textbox')).toHaveClass('d-none')
    expect(screen.getByText('Nothing to preview')).toBeInTheDocument()
  })

  test('Should display markdown preview text when switch to Preview mode with input', async () => {
    const { container } = render(<TestComponent />)

    const textbox = screen.getByRole('textbox')
    await userEvent.click(textbox)
    await userEvent.type(textbox, 'Some **Text**')

    await userEvent.click(screen.getByRole('button', { name: 'Preview' }))

    expect(screen.getByRole('textbox')).toHaveClass('d-none')
    expect(screen.queryByTestId('markdown')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  test('Should save cursor state going to and back from preview', async () => {
    render(<TestComponent />)
    const textbox = screen.getByRole('textbox')

    await userEvent.click(textbox)
    await userEvent.type(textbox, 'Hello,{enter}World!')
    expect(textbox).toHaveValue('Hello,\nWorld!')

    await userEvent.click(screen.getByRole('button', { name: 'Preview' }))
    expect(textbox).toHaveClass('d-none')

    await userEvent.click(screen.getByRole('button', { name: 'Write' }))
    await userEvent.type(textbox, '{backspace}Tom!', {
      initialSelectionStart: 7,
      initialSelectionEnd: 13,
      skipClick: false
    })
    expect(textbox).toHaveValue('Hello,\nTom!')
  })
  test('Should automatically resize to fit content', () => {
    render(<TestComponent />)
    // autoSize function runs on first render with useEffect.
    // Only picks up '+2px' additional padding as JSDOM doesnt properly mock styles
    // but this proves the code was exercised and is good enough for unit test
    expect(screen.getByRole('textbox')).toHaveStyle('height: 2px')
  })
  test('Show not auto resize after user sets height', async () => {
    render(<TestComponent />)
    const textbox = screen.getByRole('textbox')

    jest
      .spyOn(textbox, 'clientHeight', 'get')
      .mockImplementation(() => 200)
      .mockImplementationOnce(() => 100)

    expect(textbox).not.toHaveStyle('height: 200px')
    await userEvent.click(textbox)
    expect(textbox).toHaveStyle('height: 200px')

    await userEvent.click(textbox)
    expect(textbox).toHaveStyle('height: 200px')
    await userEvent.type(
      textbox,
      'Lots{enter}{enter}{enter}{enter}o{enter}{enter}{enter}{enter}{enter}lines'
    )
    expect(textbox).toHaveStyle('height: 200px')
  })
  test('Should render toolbar when on mobile', () => {
    expect.assertions(1)

    render(<TestComponent />)

    useBreakpoint.mockImplementation(() => true)

    expect(screen.getByRole('toolbar')).toBeInTheDocument()
  })
  test('Undo should restore previous text', async () => {
    render(<TestComponent />)
    const textbox = screen.getByRole('textbox')

    await userEvent.click(textbox)
    await userEvent.type(textbox, 'Hello,{enter}Tom!!')

    expect(textbox).toHaveValue('Hello,\nTom!!')
    await userEvent.type(textbox, '{Control>}zz')
    expect(textbox).toHaveValue('Hello,\nTom')
  })
  test('Undo should stop if there is no text to undo', async () => {
    render(<TestComponent />)
    const textbox = screen.getByRole('textbox')

    await userEvent.click(textbox)
    await userEvent.type(textbox, 'Tom')

    expect(textbox).toHaveValue('Tom')
    await userEvent.type(textbox, '{Control>}zzzzzzz')
    expect(textbox).toHaveValue('')
  })
  test('Redo should restore previous text from undo', async () => {
    render(<TestComponent />)
    const textbox = screen.getByRole('textbox')

    await userEvent.click(textbox)
    await userEvent.type(textbox, 'Hello,{enter}Tom!!')

    expect(textbox).toHaveValue('Hello,\nTom!!')
    await userEvent.type(textbox, '{Control>}zz')
    expect(textbox).toHaveValue('Hello,\nTom')
    await userEvent.type(textbox, '{Control>}yy')
    expect(textbox).toHaveValue('Hello,\nTom!!')
  })
  test('Redo should stop if there is no text to redo', async () => {
    expect.assertions(3)

    render(<TestComponent />)
    const textbox = screen.getByRole('textbox')

    await userEvent.click(textbox)
    await userEvent.type(textbox, 'Tom')

    expect(textbox).toHaveValue('Tom')
    await userEvent.type(textbox, '{Control>}zzzzzzz')
    expect(textbox).toHaveValue('')
    await userEvent.type(textbox, '{Control>}yyyyyyy')
    expect(textbox).toHaveValue('Tom')
  })
  test('Undo/Redo history should be reset if state is updated outside of component', async () => {
    const TestRig = () => {
      const [value, setValue] = React.useState('')
      return (
        <div>
          <button onClick={() => setValue('Reset')}>Reset</button>
          <MdInput value={value} onChange={setValue} />
        </div>
      )
    }
    render(<TestRig />)

    const textbox = screen.getByRole('textbox')

    await userEvent.click(textbox)
    await userEvent.type(textbox, 'Tom')

    expect(textbox).toHaveValue('Tom')
    await userEvent.type(textbox, '{Control>}z')
    expect(textbox).toHaveValue('To')
    await userEvent.click(screen.getByRole('button', { name: 'Reset' }))

    expect(textbox).toHaveValue('Reset')
    await userEvent.type(textbox, '{Control>}z')
    expect(textbox).toHaveValue('Reset')
    await userEvent.type(textbox, '{Control>}y')
    expect(textbox).toHaveValue('Reset')
  })
  test('Should exercise handleMarkdown and insert bold markdown style ', async () => {
    render(<TestComponent />)
    const textbox = screen.getByRole('textbox')
    await userEvent.type(textbox, '{Control>}b')
    expect(textbox).toHaveValue('****')
  })
  test('Should exercise handleMarkdown and insert bold markdown style with mac hotkey ', async () => {
    mockUseIsMac.mockImplementationOnce(() => true)
    render(<TestComponent />)
    const textbox = screen.getByRole('textbox')
    await userEvent.type(textbox, '{meta>}b')
    expect(textbox).toHaveValue('****')
  })
})
