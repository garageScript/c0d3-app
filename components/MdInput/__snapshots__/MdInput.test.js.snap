// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MdInput Component Should display markdown preview text when switch to Preview mode with input 1`] = `
<div>
  <div>
    <div
      class="nav nav-tabs"
    >
      <div
        class="nav-item"
      >
        <a
          class="nav-link"
          data-rr-ui-event-key="Write"
          href="#"
          role="button"
          style="color: rgb(142, 142, 142);"
          tabindex="0"
        >
          Write
        </a>
      </div>
      <div
        class="nav-item"
      >
        <a
          class="nav-link active"
          data-rr-ui-event-key="Preview"
          href="#"
          role="button"
          style="color: rgb(41, 41, 41);"
          tabindex="0"
        >
          Preview
        </a>
      </div>
    </div>
    <span
      class="preview"
      data-testid="markdown"
    >
      Some 
      <strong>
        Text
      </strong>
    </span>
    <hr />
    <textarea
      class="textarea d-none"
      data-testid="textbox"
      placeholder="Type something..."
      style="height: 2px;"
    >
      Some **Text**
    </textarea>
  </div>
</div>
`;

exports[`MdInput Component Should render text onto textbox when user types 1`] = `
<div>
  <div>
    <div
      class="nav nav-tabs"
    >
      <div
        class="nav-item"
      >
        <a
          class="nav-link active"
          data-rr-ui-event-key="Write"
          href="#"
          role="button"
          style="color: rgb(41, 41, 41);"
          tabindex="0"
        >
          Write
        </a>
      </div>
      <div
        class="nav-item"
      >
        <a
          class="nav-link"
          data-rr-ui-event-key="Preview"
          href="#"
          role="button"
          style="color: rgb(142, 142, 142);"
          tabindex="0"
        >
          Preview
        </a>
      </div>
      <div
        class="ms-auto btn-toolbar"
        role="toolbar"
      >
        <button
          aria-label="Add header text"
          class="btn"
        >
          <svg
            aria-hidden="true"
            class="octicon octicon-heading"
            fill="currentColor"
            focusable="false"
            height="16"
            role="img"
            style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M3.75 2a.75.75 0 0 1 .75.75V7h7V2.75a.75.75 0 0 1 1.5 0v10.5a.75.75 0 0 1-1.5 0V8.5h-7v4.75a.75.75 0 0 1-1.5 0V2.75A.75.75 0 0 1 3.75 2Z"
            />
          </svg>
        </button>
        <button
          aria-label="Add bold text <ctrl+b>"
          class="btn"
        >
          <svg
            aria-hidden="true"
            class="octicon octicon-bold"
            fill="currentColor"
            focusable="false"
            height="16"
            role="img"
            style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M4 2h4.5a3.501 3.501 0 0 1 2.852 5.53A3.499 3.499 0 0 1 9.5 14H4a1 1 0 0 1-1-1V3a1 1 0 0 1 1-1Zm1 7v3h4.5a1.5 1.5 0 0 0 0-3Zm3.5-2a1.5 1.5 0 0 0 0-3H5v3Z"
            />
          </svg>
        </button>
        <button
          aria-label="Add italic text <ctrl+i>"
          class="btn"
        >
          <svg
            aria-hidden="true"
            class="octicon octicon-italic"
            fill="currentColor"
            focusable="false"
            height="16"
            role="img"
            style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M6 2.75A.75.75 0 0 1 6.75 2h6.5a.75.75 0 0 1 0 1.5h-2.505l-3.858 9H9.25a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.505l3.858-9H6.75A.75.75 0 0 1 6 2.75Z"
            />
          </svg>
        </button>
        <button
          aria-label="Insert a quote <ctrl+shift+.>"
          class="btn"
        >
          <svg
            aria-hidden="true"
            class="octicon octicon-quote"
            fill="currentColor"
            focusable="false"
            height="16"
            role="img"
            style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M1.75 2.5h10.5a.75.75 0 0 1 0 1.5H1.75a.75.75 0 0 1 0-1.5Zm4 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2.5 7.75v6a.75.75 0 0 1-1.5 0v-6a.75.75 0 0 1 1.5 0Z"
            />
          </svg>
        </button>
        <button
          aria-label="Insert code <ctrl+e>"
          class="btn"
        >
          <svg
            aria-hidden="true"
            class="octicon octicon-code"
            fill="currentColor"
            focusable="false"
            height="16"
            role="img"
            style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z"
            />
          </svg>
        </button>
        <button
          aria-label="Add a link <ctrl+k>"
          class="btn"
        >
          <svg
            aria-hidden="true"
            class="octicon octicon-link"
            fill="currentColor"
            focusable="false"
            height="16"
            role="img"
            style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z"
            />
          </svg>
        </button>
        <button
          aria-label="Add a bulleted list <ctrl+shift+8>"
          class="btn"
        >
          <svg
            aria-hidden="true"
            class="octicon octicon-list-unordered"
            fill="currentColor"
            focusable="false"
            height="16"
            role="img"
            style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z"
            />
          </svg>
        </button>
        <button
          aria-label="Add a numbered list <ctrl+shift+7>"
          class="btn"
        >
          <svg
            aria-hidden="true"
            class="octicon octicon-list-ordered"
            fill="currentColor"
            focusable="false"
            height="16"
            role="img"
            style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M5 3.25a.75.75 0 0 1 .75-.75h8.5a.75.75 0 0 1 0 1.5h-8.5A.75.75 0 0 1 5 3.25Zm0 5a.75.75 0 0 1 .75-.75h8.5a.75.75 0 0 1 0 1.5h-8.5A.75.75 0 0 1 5 8.25Zm0 5a.75.75 0 0 1 .75-.75h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1-.75-.75ZM.924 10.32a.5.5 0 0 1-.851-.525l.001-.001.001-.002.002-.004.007-.011c.097-.144.215-.273.348-.384.228-.19.588-.392 1.068-.392.468 0 .858.181 1.126.484.259.294.377.673.377 1.038 0 .987-.686 1.495-1.156 1.845l-.047.035c-.303.225-.522.4-.654.597h1.357a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5c0-1.005.692-1.52 1.167-1.875l.035-.025c.531-.396.8-.625.8-1.078a.57.57 0 0 0-.128-.376C1.806 10.068 1.695 10 1.5 10a.658.658 0 0 0-.429.163.835.835 0 0 0-.144.153ZM2.003 2.5V6h.503a.5.5 0 0 1 0 1H.5a.5.5 0 0 1 0-1h.503V3.308l-.28.14a.5.5 0 0 1-.446-.895l1.003-.5a.5.5 0 0 1 .723.447Z"
            />
          </svg>
        </button>
      </div>
    </div>
    <textarea
      class="textarea"
      data-testid="textbox"
      placeholder="Type something..."
      style="height: 2px;"
    >
      Javascript
    </textarea>
  </div>
</div>
`;

exports[`MdInput Component Should switch to Write mode when user clicks Write button 1`] = `
<div>
  <div>
    <div
      class="nav nav-tabs"
    >
      <div
        class="nav-item"
      >
        <a
          class="nav-link active"
          data-rr-ui-event-key="Write"
          href="#"
          role="button"
          style="color: rgb(41, 41, 41);"
          tabindex="0"
        >
          Write
        </a>
      </div>
      <div
        class="nav-item"
      >
        <a
          class="nav-link"
          data-rr-ui-event-key="Preview"
          href="#"
          role="button"
          style="color: rgb(142, 142, 142);"
          tabindex="0"
        >
          Preview
        </a>
      </div>
      <div
        class="ms-auto btn-toolbar"
        role="toolbar"
      >
        <button
          aria-label="Add header text"
          class="btn"
        >
          <svg
            aria-hidden="true"
            class="octicon octicon-heading"
            fill="currentColor"
            focusable="false"
            height="16"
            role="img"
            style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M3.75 2a.75.75 0 0 1 .75.75V7h7V2.75a.75.75 0 0 1 1.5 0v10.5a.75.75 0 0 1-1.5 0V8.5h-7v4.75a.75.75 0 0 1-1.5 0V2.75A.75.75 0 0 1 3.75 2Z"
            />
          </svg>
        </button>
        <button
          aria-label="Add bold text <ctrl+b>"
          class="btn"
        >
          <svg
            aria-hidden="true"
            class="octicon octicon-bold"
            fill="currentColor"
            focusable="false"
            height="16"
            role="img"
            style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M4 2h4.5a3.501 3.501 0 0 1 2.852 5.53A3.499 3.499 0 0 1 9.5 14H4a1 1 0 0 1-1-1V3a1 1 0 0 1 1-1Zm1 7v3h4.5a1.5 1.5 0 0 0 0-3Zm3.5-2a1.5 1.5 0 0 0 0-3H5v3Z"
            />
          </svg>
        </button>
        <button
          aria-label="Add italic text <ctrl+i>"
          class="btn"
        >
          <svg
            aria-hidden="true"
            class="octicon octicon-italic"
            fill="currentColor"
            focusable="false"
            height="16"
            role="img"
            style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M6 2.75A.75.75 0 0 1 6.75 2h6.5a.75.75 0 0 1 0 1.5h-2.505l-3.858 9H9.25a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.505l3.858-9H6.75A.75.75 0 0 1 6 2.75Z"
            />
          </svg>
        </button>
        <button
          aria-label="Insert a quote <ctrl+shift+.>"
          class="btn"
        >
          <svg
            aria-hidden="true"
            class="octicon octicon-quote"
            fill="currentColor"
            focusable="false"
            height="16"
            role="img"
            style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M1.75 2.5h10.5a.75.75 0 0 1 0 1.5H1.75a.75.75 0 0 1 0-1.5Zm4 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2.5 7.75v6a.75.75 0 0 1-1.5 0v-6a.75.75 0 0 1 1.5 0Z"
            />
          </svg>
        </button>
        <button
          aria-label="Insert code <ctrl+e>"
          class="btn"
        >
          <svg
            aria-hidden="true"
            class="octicon octicon-code"
            fill="currentColor"
            focusable="false"
            height="16"
            role="img"
            style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z"
            />
          </svg>
        </button>
        <button
          aria-label="Add a link <ctrl+k>"
          class="btn"
        >
          <svg
            aria-hidden="true"
            class="octicon octicon-link"
            fill="currentColor"
            focusable="false"
            height="16"
            role="img"
            style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z"
            />
          </svg>
        </button>
        <button
          aria-label="Add a bulleted list <ctrl+shift+8>"
          class="btn"
        >
          <svg
            aria-hidden="true"
            class="octicon octicon-list-unordered"
            fill="currentColor"
            focusable="false"
            height="16"
            role="img"
            style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z"
            />
          </svg>
        </button>
        <button
          aria-label="Add a numbered list <ctrl+shift+7>"
          class="btn"
        >
          <svg
            aria-hidden="true"
            class="octicon octicon-list-ordered"
            fill="currentColor"
            focusable="false"
            height="16"
            role="img"
            style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M5 3.25a.75.75 0 0 1 .75-.75h8.5a.75.75 0 0 1 0 1.5h-8.5A.75.75 0 0 1 5 3.25Zm0 5a.75.75 0 0 1 .75-.75h8.5a.75.75 0 0 1 0 1.5h-8.5A.75.75 0 0 1 5 8.25Zm0 5a.75.75 0 0 1 .75-.75h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1-.75-.75ZM.924 10.32a.5.5 0 0 1-.851-.525l.001-.001.001-.002.002-.004.007-.011c.097-.144.215-.273.348-.384.228-.19.588-.392 1.068-.392.468 0 .858.181 1.126.484.259.294.377.673.377 1.038 0 .987-.686 1.495-1.156 1.845l-.047.035c-.303.225-.522.4-.654.597h1.357a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5c0-1.005.692-1.52 1.167-1.875l.035-.025c.531-.396.8-.625.8-1.078a.57.57 0 0 0-.128-.376C1.806 10.068 1.695 10 1.5 10a.658.658 0 0 0-.429.163.835.835 0 0 0-.144.153ZM2.003 2.5V6h.503a.5.5 0 0 1 0 1H.5a.5.5 0 0 1 0-1h.503V3.308l-.28.14a.5.5 0 0 1-.446-.895l1.003-.5a.5.5 0 0 1 .723.447Z"
            />
          </svg>
        </button>
      </div>
    </div>
    <textarea
      class="textarea"
      data-testid="textbox"
      placeholder="Type something..."
      style="height: 2px;"
    />
  </div>
</div>
`;
